index.html

```diff
@@ -10198,127 +10215,229 @@ <h2>User prompts</h2>
+
+<p>A <dfn>prompt handler configuration</dfn> is a <a>struct</a> with two
+items; a <dfn data-dfn-for="prompt handler configuration">handler</dfn>, which is a
+string, and a <dfn data-dfn-for="prompt handler configuration">notify</dfn>,
+which is a boolean.
 
-<p>The <dfn>current user prompt</dfn> is said to be the active <a>user prompt</a>,
- which can be one of the entries on the <a>table of simple dialogs</a>.
+<p>To <dfn>serialize a prompt handler configuration</dfn>
+given <var>configuration</var>:
 
-<p>To <dfn data-lt="dismissed|dismisses">dismiss</dfn>
- the <a>current user prompt</a>,
- do so as if the user would click the <b>Cancel</b> or <b>OK</b> button,
- whichever is present, in that order.
+<ol class=algorithm>
 
-<p>To <dfn data-lt="accepting|accepted|accepts">accept</dfn> the <a>current user prompt</a>,
- do so as if the user would click the <b>OK</b> button.
+ <li><p>Let <var>serialized</var>
+ be <var>configuration</var>&apos;s [=prompt handler
+ configuration/handler=].
 
-<p>
-The <dfn>user prompt handler</dfn> defines what action
-the <a>remote end</a> must take when a <a>user prompt</a> is encountered.
-This is defined by the <a>unhandled prompt behavior</a> capability.
-The following <dfn>known prompt handling approaches table</dfn>
-lists the keywords and states for the attribute:
+ <li><p>If «"<code>dismiss</code>", "<code>accept</code>"»
+ [=list/contains=] <var>serialized</var>,
+ and <var>configuration</var>&apos;s [=prompt handler
+ configuration/notify=] is true, append "<code> and notify</code>"
+ to <var>serialized</var>.
 
+ <li><p>Return <var>serialized</var>.
+
+</ol>
+
+The <dfn>known prompt handlers</dfn> are:
 <table class=simple>
  <tr>
-  <th>Keyword
-  <th>State
-  <th>Description
+  <th>Handler key
+  <th>Description (non-normative).
  </tr>
 
  <tr>
   <td>"<code>dismiss</code>"
-  <td><dfn>Dismiss state</dfn>
   <td>All <a>simple dialogs</a> encountered should be <a>dismissed</a>.
  </tr>
 
  <tr>
   <td>"<code>accept</code>"
-  <td><dfn>Accept state</dfn>
   <td>All <a>simple dialogs</a> encountered should be <a>accepted</a>.
  </tr>
 
  <tr>
   <td>"<code>dismiss and notify</code>"
-  <td><dfn>Dismiss and notify state</dfn>
   <td>All <a>simple dialogs</a> encountered should be <a>dismissed</a>,
       and an error returned that the dialog was handled.
  </tr>
 
  <tr>
   <td>"<code>accept and notify</code>"
-  <td><dfn>Accept and notify state</dfn>
   <td>All <a>simple dialogs</a> encountered should be <a>accepted</a>,
       and an error returned that the dialog was handled.
  </tr>
 
  <tr>
   <td>"<code>ignore</code>"
-  <td><dfn>Ignore state</dfn>
   <td>All <a>simple dialogs</a> encountered should be left to the user to handle.
  </tr>
 </table>
 
-<p>When required to <dfn>deserialize as an unhandled prompt behavior</dfn> an
+<p>The <dfn>valid prompt types</dfn> are «"<code>alert</code>",
+"<code>beforeUnload</code>", "<code>confirm</code>",
+"<code>default</code>", "<code>prompt</code>"».
```

Comment:
Maybe add a note that explains what a `default` prompt type is used for in WebDriver?